### PR TITLE
Qt/FIFOPlayer: Implement Analyzer

### PR DESCRIPTION
**Pull request by spycrab**
Screenshot

(Also features splitters so you can freely resize everything to your liking!)

### CODE REVIEW

**Comment by JMC47** [comment]
@booto do you want to check this out as well?

**Comment by JMC47** [comment]
Seems to work correctly!  I don't know what a lot of the numbers mean or do, so I can't really sanity check it.

File: Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+      break;
+
+      case OpcodeDecoder::GX_LOAD_INDX_A:
+      case OpcodeDecoder::GX_LOAD_INDX_B:
+      case OpcodeDecoder::GX_LOAD_INDX_C:
+      case OpcodeDecoder::GX_LOAD_INDX_D:
+      {
+        objectdata += 4;
+        new_label = QStringLiteral("LOAD INDX %1")
+                        .arg((command == OpcodeDecoder::GX_LOAD_INDX_A) ?
```
[the remark this answers is not in the document]
Comment (spycrab) [reply Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp]: I dunno. I should probably get rid of it altogether tbh. Not sure what to replace it with yet. (Copied from Wx btw).

File: Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+
+  int frame_nr = items[0]->data(0, FRAME_ROLE).toInt();
+  int object_nr = items[0]->data(0, OBJECT_ROLE).toInt();
+
+  const auto& frame_info = FifoPlayer::GetInstance().GetAnalyzedFrameInfo(frame_nr);
+  const auto& fifo_frame = FifoPlayer::GetInstance().GetFile()->GetFrame(frame_nr);
+
+  const u8* objectdata_start = &fifo_frame.fifoData[frame_info.objectStarts[object_nr]];
+  const u8* objectdata_end = &fifo_frame.fifoData[frame_info.objectEnds[object_nr]];
+  const u8* objectdata = static_cast<const u8*>(objectdata_start);
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp]: What's this cast needed for?

File: Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+  const int obj_offset = objectdata_start - &fifo_frame.fifoData[frame_info.objectStarts[0]];
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp]: `std::ptrdiff_t`

File: Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+                        .arg(cmd2 & 0xFFFFFF, 6, 16, QLatin1Char('0'));
+      }
+      break;
+
+      default:
+        new_label = tr("Unexpected 0x80 call? Aborting...");
+        objectdata = static_cast<const u8*>(next_objdata_start);
+        break;
+      }
+      new_label = QStringLiteral("%1:  ").arg(new_offset, 8, 16, QLatin1Char('0')) + new_label;
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp]: Are those two spaces intentional (for alignment or whatever)?

Comment (spycrab) [reply Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp]: Yep.

**Pull request merged**